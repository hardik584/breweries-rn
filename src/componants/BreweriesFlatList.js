import React from "react";
import {
    FlatList,
    Text,
    View,
    TouchableOpacity,
  } from 'react-native';

import CustomStyle from '../CustomStyle';

const BreweriesFlatList = (props) => {
  return (
    <FlatList
    data={props.data}
    keyExtractor={({id}, index) => index}
    renderItem={({item, index}) => (
      <TouchableOpacity
        onPress={() =>
            props.navigation.navigate('BreweriesDetails', {
            breweriesItem: item,
          })
        }>
        <View
          style={[
            CustomStyle.detailsBackgroundStyle,
            {backgroundColor: index % 2 == 0 ? '#ddd' : 'white'},
          ]}>
          <View style={CustomStyle.viewRowStyle}>
            <View style={CustomStyle.circleShape}>
              <Text style={{align: 'center'}}> {index + 1} </Text>
            </View>
            <Text style={CustomStyle.baseText}>{item.name} ðŸ‘†</Text>

          </View>
          <Text>
            {item.street}
            {item.city}, {item.state}, {item.country},{' '}
            {item.postal_code}
          </Text>
        </View>
      </TouchableOpacity>
    )}
  />
  );
}
export default BreweriesFlatList;
