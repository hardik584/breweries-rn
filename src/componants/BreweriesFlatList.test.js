import BreweriesFlatList from './BreweriesFlatList';
// import BreweriesDeatils from "../BreweriesDeatils";
import renderer from 'react-test-renderer';
import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render, screen, fireEvent} from '@testing-library/react-native';

import AppNavigator from '../App';

it(`Checking Breweries FlatList Rendering`, () => {
  let data = [
    {
      id: 'alphabet-city-brewing-co-new-york',
      name: 'Alphabet City Brewing Co',
      brewery_type: 'contract',
      street: '96 Avenue C Frnt 4',
      address_2: null,
      address_3: null,
      city: 'New York',
      state: 'New York',
      county_province: null,
      postal_code: '10009-7055',
      country: 'United States',
      longitude: null,
      latitude: null,
      phone: '9143566809',
      website_url: 'http://www.acbnyc.com',
      updated_at: '2022-08-20T02:56:08.975Z',
      created_at: '2022-08-20T02:56:08.975Z',
    },
  ];
  const tree = renderer.create(
    <BreweriesFlatList data={data}> </BreweriesFlatList>,
  );
  expect(tree).toMatchSnapshot();
});

describe('Home', () => {
  const fakeNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };
  it(`Checking Breweries FlatList Rendering`, () => {
    let data = [
      {
        id: 'alphabet-city-brewing-co-new-york',
        name: 'Alphabet City Brewing Co',
        brewery_type: 'contract',
        street: '96 Avenue C Frnt 4',
        address_2: null,
        address_3: null,
        city: 'New York',
        state: 'New York',
        county_province: null,
        postal_code: '10009-7055',
        country: 'United States',
        longitude: null,
        latitude: null,
        phone: '9143566809',
        website_url: 'http://www.acbnyc.com',
        updated_at: '2022-08-20T02:56:08.975Z',
        created_at: '2022-08-20T02:56:08.975Z',
      },
    ];
    const tree = renderer
      .create(
        <BreweriesFlatList data={data} navigation={fakeNavigation}>
          {' '}
        </BreweriesFlatList>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
