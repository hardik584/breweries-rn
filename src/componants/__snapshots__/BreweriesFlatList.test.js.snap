// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Breweries FlatList Rendering 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "address_2": null,
        "address_3": null,
        "brewery_type": "contract",
        "city": "New York",
        "country": "United States",
        "county_province": null,
        "created_at": "2022-08-20T02:56:08.975Z",
        "id": "alphabet-city-brewing-co-new-york",
        "latitude": null,
        "longitude": null,
        "name": "Alphabet City Brewing Co",
        "phone": "9143566809",
        "postal_code": "10009-7055",
        "state": "New York",
        "street": "96 Avenue C Frnt 4",
        "updated_at": "2022-08-20T02:56:08.975Z",
        "website_url": "http://www.acbnyc.com",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#38b6ba",
                  "borderRadius": 75,
                  "height": 30,
                  "justifyContent": "center",
                  "marginRight": 5,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "align": "center",
                  }
                }
              >
                 
                1
                 
              </Text>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Alphabet City Brewing Co
               ðŸ‘†
            </Text>
          </View>
          <Text>
            96 Avenue C Frnt 4
            New York
            , 
            New York
            , 
            United States
            ,
             
            10009-7055
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Home Checking Breweries FlatList Rendering 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "address_2": null,
        "address_3": null,
        "brewery_type": "contract",
        "city": "New York",
        "country": "United States",
        "county_province": null,
        "created_at": "2022-08-20T02:56:08.975Z",
        "id": "alphabet-city-brewing-co-new-york",
        "latitude": null,
        "longitude": null,
        "name": "Alphabet City Brewing Co",
        "phone": "9143566809",
        "postal_code": "10009-7055",
        "state": "New York",
        "street": "96 Avenue C Frnt 4",
        "updated_at": "2022-08-20T02:56:08.975Z",
        "website_url": "http://www.acbnyc.com",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#ddd",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#38b6ba",
                  "borderRadius": 75,
                  "height": 30,
                  "justifyContent": "center",
                  "marginRight": 5,
                  "width": 30,
                }
              }
            >
              <Text
                style={
                  Object {
                    "align": "center",
                  }
                }
              >
                 
                1
                 
              </Text>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Alphabet City Brewing Co
               ðŸ‘†
            </Text>
          </View>
          <Text>
            96 Avenue C Frnt 4
            New York
            , 
            New York
            , 
            United States
            ,
             
            10009-7055
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
